jupyterhub_config.py

```diff
@@ -242,18 +244,22 @@ def finalize_user_content(spawner):
 # Grab email, lab info on all users
 ff_users = ff_utils.search_metadata('search/?type=User&field=email&field=lab', key=ff_keys)
 for ff_user in ff_users:
-
+    email, lab = ff_user.get('email'), ff_user.get('lab')
     # Allow users into JH if they have both email and lab fields
-    if not ff_user.get('email') or not ff_user.get('lab'):
+    if not email or not lab:
+        print(f'Blocking user {email}')
+        blocked_users.add(email)
         continue
-    allowed_users.add(ff_user['email'])
+    else:
+        allowed_users.add(email)
 
-    # If first admin email from before is located, add it to admin set
-    if ff_user['email'].lower() in admin_emails:
-        admin.add(ff_user['email'])
+        # If first admin email from before is located, add it to admin set
+        if email.lower() in admin_emails:
+            admin.add(email)
```

Comment:
You're trying to add the non-lowercased email to a list you'll later check by using a lowered name.
```suggestion
        # If first admin email from before is located, add it to admin set
        email = email.lower()
        if email in admin_emails:
            admin.add(email)
```